### PR TITLE
layerscape: improve support of two LS1012A boards

### PR DESCRIPTION
**Pull request by CHKDSK88**
This PR improve support of two LS1012A boards. One commit add fit-packed initramfs images. It make run test images easier.

### CODE REVIEW

**Comment by adschm** [comment]
> "This will make developing process easier."

Why?/How? -> Commit message

File: target/linux/layerscape/patches-5.4/900-layerscape_fix_support_of_LS1012A-RDB.patch [hunk 1]
```diff
@@ -0,0 +1,91 @@
+--- a/arch/arm64/boot/dts/freescale/fsl-ls1012a-rdb.dts
++++ b/arch/arm64/boot/dts/freescale/fsl-ls1012a-rdb.dts
+@@ -6,7 +6,8 @@
+  *
+  */
+ /dts-v1/;
+-
```
Comment (adschm) [inline target/linux/layerscape/patches-5.4/900-layerscape_fix_support_of_LS1012A-RDB.patch:7]: Nitpick: I would have just kept this empty line.

**Comment by CHKDSK88** [comment]
> 
> 
> > "This will make developing process easier."
> 
> Why?/How? -> Commit message

Because initramfs image will contain all data required to start image.

File: target/linux/layerscape/patches-5.4/900-layerscape_fix_support_of_LS1012A-RDB.patch (hunk 1 above)
Comment (CHKDSK88) [reply target/linux/layerscape/patches-5.4/900-layerscape_fix_support_of_LS1012A-RDB.patch]: Done.

**Comment by CHKDSK88** [comment]
@yangbolu1991 

Could You review it?

File: package/boot/uboot-layerscape/files/fsl_ls1012a-rdb-uEnv.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ fdtaddr=0x8f000000
 loadaddr=0x81000000
 fdt_high=0xffffffffffffffff
 initrd_high=0xffffffffffffffff
```
Comment (yangbolu1991) [inline package/boot/uboot-layerscape/files/fsl_ls1012a-rdb-uEnv.txt:4]: fdt_high/initrd_high should be removed because of introducing bootm_size.

**Comment by yangbolu1991** [comment]
Other changes look good to me.

Some suggestions on patch commit message :)
1. Describe what the patch actually did in title. This makes user understand the patch when seeing just the title.
> layerscape: improve support of LS1012A-RDB
like, layerscape: enable LS1012ARDB I2C peripherals
2. Add commit message for patches which are put under kernel and package directory, than putting just changes diff.

Thanks.

File: package/firmware/layerscape/ls-rcw/patches/0002-fix_rcw_for_ls1012a-frdm.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+--- a/ls1012afrdm/N_SSNP_3305/rcw_800.rcw
++++ b/ls1012afrdm/N_SSNP_3305/rcw_800.rcw
```
Comment (adschm) [inline package/firmware/layerscape/ls-rcw/patches/0002-fix_rcw_for_ls1012a-frdm.patch:2]: Please bump PKG_RELEASE for the ls-rcw package.

**Comment by adschm** [comment]
The "improve support of" commit title could be made more specific by just telling what you do to improve support, e.g. replace "layerscape: improve support of two LS1012A boards" by "layerscape: enable i2c peripherials on two LS1012A boards" or similar

File: package/boot/uboot-layerscape/files/fsl_ls1012a-rdb-uEnv.txt (hunk 2 above)
Comment (CHKDSK88) [reply package/boot/uboot-layerscape/files/fsl_ls1012a-rdb-uEnv.txt]: Done. Thank You.

File: package/firmware/layerscape/ls-rcw/patches/0002-fix_rcw_for_ls1012a-frdm.patch (hunk 3 above)
Comment (CHKDSK88) [reply package/firmware/layerscape/ls-rcw/patches/0002-fix_rcw_for_ls1012a-frdm.patch:2]: Done.

**Comment by CHKDSK88** [comment]
@adschm 
@yangbolu1991 

I made requested changes. Please look if now descriptions are ok.

**Comment by yangbolu1991** [comment]
It looks fine to me.
Thanks:)

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Comment by CHKDSK88** [comment]
@ynezz 
Thank You.

@yangbolu1991
Are You going to send patches from this PR to NXP kernel? Or should I port it to upstream?

**Comment by yangbolu1991** [comment]
@CHKDSK88 
You may send to upstream.
Current Layerscape SDK is in code freeze for coming LSDK-20.12 release. The next release after it will be next year.
Thanks.

**Comment by CHKDSK88** [comment]
I was sent upstream:
http://lists.infradead.org/pipermail/linux-arm-kernel/2021-January/631627.html
http://lists.infradead.org/pipermail/linux-arm-kernel/2021-January/631628.html

**Pull request closed without merging**